fast-rtps 1.9.0 does not build without -DSECURITY=ON [6003]

When 1.9.0 (1b787e2d0) building normally (no extra CMake options) I get this error:

```
In file included from /Users/william/ros2_ws/src/eProsima/Fast-RTPS/src/cpp/rtps/messages/RTPSMessageGroup.cpp:20:
/Users/william/ros2_ws/src/eProsima/Fast-RTPS/include/fastrtps/rtps/writer/../messages/RTPSMessageGroup.h:209:30: warning: private field 'participant_' is not used [-Wunused-private-field]
        RTPSParticipantImpl* participant_;
                             ^
1 warning generated.
/Users/william/ros2_ws/src/eProsima/Fast-RTPS/src/cpp/rtps/builtin/discovery/participant/PDPClient.cpp:95:38: error: no member named 'is_secure' in 'eprosima::fastrtps::rtps::RTPSParticipantImpl'
        builtin->mp_participantImpl->is_secure())
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~  ^
/Users/william/ros2_ws/src/eProsima/Fast-RTPS/src/cpp/rtps/builtin/discovery/participant/PDPServer.cpp:64:36: error: no member named 'is_secure' in 'eprosima::fastrtps::rtps::RTPSParticipantImpl'
        built->mp_participantImpl->is_secure())
        ~~~~~~~~~~~~~~~~~~~~~~~~~  ^
1 error generated.
make[2]: *** [src/cpp/CMakeFiles/fastrtps.dir/rtps/builtin/discovery/participant/PDPClient.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
1 error generated.
make[2]: *** [src/cpp/CMakeFiles/fastrtps.dir/rtps/builtin/discovery/participant/PDPServer.cpp.o] Error 1
make[1]: *** [src/cpp/CMakeFiles/fastrtps.dir/all] Error 2
make: *** [all] Error 2
```

I do not have this issue if I turn security on with the CMake option `-DSECURITY=ON` (and also since I'm on macOS I have to also do `export OPENSSL_ROOT_DIR=$(brew --prefix openssl)`).

--- a/include/fastrtps/rtps/messages/RTPSMessageGroup.h
+++ b/include/fastrtps/rtps/messages/RTPSMessageGroup.h
@@ -206,8 +206,6 @@ class RTPSWriter;
 
         const RTPSMessageSenderInterface& sender_;
             
-        RTPSParticipantImpl* participant_;
-
         Endpoint* endpoint_;
 
         CDRMessage_t* full_msg_;
@@ -219,6 +217,8 @@ class RTPSWriter;
         GuidPrefix_t current_dst_;
 
 #if HAVE_SECURITY
+        RTPSParticipantImpl* participant_;
+
         CDRMessage_t* encrypt_msg_;
 #endif
 

--- a/src/cpp/rtps/builtin/discovery/participant/PDPClient.cpp
+++ b/src/cpp/rtps/builtin/discovery/participant/PDPClient.cpp
@@ -92,7 +92,11 @@ PDPClient::PDPClient(
     , _msgbuffer(
         DISCOVERY_PARTICIPANT_DATA_MAX_SIZE,
         builtin->mp_participantImpl->getGuid().guidPrefix,
+#if HAVE_SECURITY
         builtin->mp_participantImpl->is_secure())
+#else
+        false)
+#endif
     , mp_sync(nullptr)
     , _serverPing(false)
 {

--- a/src/cpp/rtps/builtin/discovery/participant/PDPServer.cpp
+++ b/src/cpp/rtps/builtin/discovery/participant/PDPServer.cpp
@@ -53,15 +53,19 @@ namespace fastrtps{
 namespace rtps {
 
 PDPServer::PDPServer(
-        BuiltinProtocols* built,
+        BuiltinProtocols* builtin,
         const RTPSParticipantAllocationAttributes& allocation,
         DurabilityKind_t durability_kind)
-    : PDP(built, allocation)
+    : PDP(builtin, allocation)
     , _durability(durability_kind)
     , _msgbuffer(
         DISCOVERY_PARTICIPANT_DATA_MAX_SIZE,
-        built->mp_participantImpl->getGuid().guidPrefix,
-        built->mp_participantImpl->is_secure())
+        builtin->mp_participantImpl->getGuid().guidPrefix,
+#if HAVE_SECURITY
+        builtin->mp_participantImpl->is_secure())
+#else
+        false)
+#endif
     , mp_sync(nullptr)
     , PDP_callback_(false)
 {

--- a/src/cpp/rtps/messages/RTPSMessageGroup.cpp
+++ b/src/cpp/rtps/messages/RTPSMessageGroup.cpp
@@ -155,6 +155,9 @@ RTPSMessageGroup::RTPSMessageGroup(
 #endif
     , max_blocking_time_point_(max_blocking_time_point)
 {
+    // Avoid warning when neither SECURITY nor DEBUG is used
+    (void)participant;
+
     assert(participant);
     assert(endpoint);
 

--- a/src/cpp/rtps/messages/RTPSMessageGroup.cpp
+++ b/src/cpp/rtps/messages/RTPSMessageGroup.cpp
@@ -145,12 +145,12 @@ RTPSMessageGroup::RTPSMessageGroup(
         const RTPSMessageSenderInterface& msg_sender,
         std::chrono::steady_clock::time_point max_blocking_time_point)
     : sender_(msg_sender)
-    , participant_(participant)
     , endpoint_(endpoint)
     , full_msg_(&msg_group.rtpsmsg_fullmsg_)
     , submessage_msg_(&msg_group.rtpsmsg_submessage_)
     , currentBytesSent_(0)
 #if HAVE_SECURITY
+    , participant_(participant)
     , encrypt_msg_(&msg_group.rtpsmsg_encrypt_)
 #endif
     , max_blocking_time_point_(max_blocking_time_point)

